airflow/utils/log/secrets_masker.py

```diff
@@ -0,0 +1,221 @@
+
+    if name and settings.HIDE_SENSITIVE_VAR_CONN_FIELDS:
+        name = name.strip().lower()
+        return any(s in name for s in get_sensitive_variables_fields())
+    return False
+
+
+def mask_secret(secret: Union[str, dict, Iterable], name: str = None) -> None:
+    """
+    Mask a secret from appearing in the task logs.
+
+    If ``name`` is provided, then it will only be masked if the name matches
+    one of the configured "sensitive" names.
+
+    If ``secret`` is a dict or a iterable (excluding str) then it will be
+    recursively walked and keys with sensitive names will be hidden.
+    """
+    # Delay import
+    from airflow import settings
+
+    # Filtering all log messages is not a free process, so we only do it when
+    # running tasks
+    if not settings.MASK_SECRETS_IN_LOGS or not secret:
+        return
+
+    _secrets_masker().add_mask(secret, name)
+
+
+def redact(value: "RedactableItem", name: str = None) -> "RedactableItem":
+    """Redact any secrets found in ``value``."""
+    return _secrets_masker().redact(value, name)
+
+
+@cache
+def _secrets_masker() -> "SecretsMasker":
+
+    for flt in logging.getLogger('airflow.task').filters:
+        if isinstance(flt, SecretsMasker):
+            return flt
+    raise RuntimeError("No SecretsMasker found!")
+
+
+class SecretsMasker(logging.Filter):
+    """Redact secrets from logs"""
+
+    replacer: Optional["RePatternType"] = None
+    patterns: Set[str]
+
+    ALREADY_FILTERED_FLAG = "__SecretsMasker_filtered"
+
+    def __init__(self):
+        super().__init__()
+        self.patterns = set()
+
+    @cached_property
+    def _record_attrs_to_ignore(self) -> Iterable[str]:
+        # Doing log.info(..., extra={'foo': 2}) sets extra properties on
+        # record, i.e. record.foo. And we need to filter those too. Fun
+        #
+        # Create a record, and look at what attributes are on it, and ignore
+        # all the default ones!
+
+        record = logging.getLogRecordFactory()(
+            # name, level, pathname, lineno, msg, args, exc_info, func=None, sinfo=None,
+            "x",
+            logging.INFO,
+            __file__,
+            1,
+            "",
+            tuple(),
+            exc_info=None,
+            func="funcname",
+        )
+        return frozenset(record.__dict__).difference({'msg', 'args'})
+
+    def filter(self, record) -> bool:
+        if self.ALREADY_FILTERED_FLAG in record.__dict__:
+            # Filters are attached to multiple handlers and logs, keep a
+            # "private" flag that stops us needing to process it more than once
+            return True
+
+        if self.replacer:
+            for k, v in record.__dict__.items():
+                if k in self._record_attrs_to_ignore:
+                    continue
+                record.__dict__[k] = self.redact(v)
+            if record.exc_info:
+                exc = record.exc_info[1]
+                # I'm not sure if this is a good idea!
```

Comment:
Did you wanted to think about this more before you merged this PR, or just a future TODO

Reply:
Future Todo/pointer to anyone finding this a problem in the future